### ISSUE TITLE
RFE: Reproduce Exception *Type* on Error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [etorreborre/specs#59]
```
Currently, if an expectation fails with an exception (other than a
FailureException), that exception will be caught and its message and stack
trace will be printed.  However, the actual type of the exception will not
be reproduced.  This can make a large number of exceptions very difficult
to debug, given that it is standard practice to describe the exception
entirely by the type and leave the message null (e.g. NullPointerException,
StackOverflowError).  Some other exceptions can also be *difficult* to
debug even when their messages are non-null (e.g. IllegalArgumentException).

Such exceptions (when caught by specs) should be printed with their types,
message and stack trace entirely intact.  As mentioned, this is very-nearly
the current behavior, but lacking the type.
```

Original issue reported on code.google.com by `djspie...@gmail.com` on 24 Feb 2009 at 7:35

**Comment by GoogleCodeExporter** [comment]
Original comment by `etorrebo...@gmail.com` on 24 Feb 2009 at 9:41
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
fixed in 1.4.4-SNAPSHOT. Please confirm if it's ok with you. Thanks.
```

Original comment by `etorrebo...@gmail.com` on 25 Feb 2009 at 2:27
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `etorrebo...@gmail.com` on 14 Feb 2010 at 2:43
- Changed state: **Verified**
